packages/ckeditor5-core/tests/editor/editorui.js

```diff
@@ -202,4 +202,32 @@ describe( 'EditorUI', () => {
 			sinon.assert.calledWithMatch( stub, 'editor-ui-deprecated-editable-elements' );
 		} );
 	} );
+
+	describe( '_viewportOffset()', () => {
```

Comment:
If it's private it must not be accessed by other classes. See https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/code-style.html#visibility-levels.

Reply:
Also, I didn't yet see tests, but do we have an integration test for the warning? I can see the second test here, but what if `_viewportOffset` is never accessed for some reason? Who ensures that it is?

Reply:
I changed it to private method, and updated tests to test against `ui.viewportOffset` instead of this private method.

> Also, I didn't yet see tests, but do we have an integration test for the warning?

Yes: [here](https://github.com/ckeditor/ckeditor5/blob/b609fcfa4502eeb3bfd28aec115f5f6c0121d0a6/packages/ckeditor5-editor-classic/tests/classiceditorui.js#L89) and [here](https://github.com/ckeditor/ckeditor5/blob/b609fcfa4502eeb3bfd28aec115f5f6c0121d0a6/packages/ckeditor5-editor-inline/tests/inlineeditorui.js#L88).

I updated both of these tests to check for

```
sinon.assert.calledWithMatch( spy, 'editor-ui-deprecated-viewport-offset-config' );
```

instead of just checking for `calledOnce` for easier searching.

The only thing is that I could potentially extract it to a separate case, but it just felt natural to test this warning in one go.

> I can see the second test here, but what if _viewportOffset is never accessed for some reason? Who ensures that it is?

I have troubles finding any possible scenario when it's not accessed. Maybe after my last update to the code this became obsolete.

We do set `viewportOffset` inside `EditorUI` constructor. So every editor that has an UI will call it. If editor has no UI we don't care about setting offset data, nor warning about deprecated property usage.